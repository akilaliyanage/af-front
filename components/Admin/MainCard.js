import React, {useEffect, useState} from "react";
import '../../assets/css/admin/admin.css'
import MainCardTitle from "./MainCardContent/Title";
import MainCardDate from "./MainCardContent/Date";
import axios from "axios";
import config from "../../config.json";

function Reviewers(){

    const [visible,setVisible] = useState("");
    const [result,setResult] = useState();

    useEffect( () => {

        const url = config.host+"/edi-noti";
        axios.get(url).then((res) => {
            const status = res.data;
            setResult(status);

            if(result.length === 0){
                setVisible("none");
            }


        }).catch((err) => {
            console.log(err);
        })

    })



    return(

        <div className="uditha-main-card uditha-main-card-align">
            <div className="uditha-edit-content">


           <MainCardTitle/>
                <br/>
           <MainCardDate/>

                <br/>

                <h2 style={{color:"white"}}>@</h2>
                <h1 style={{color:"white"}}>SLIIT</h1>
                <br/>
                <button style={{display:visible }} onClick={e =>  window.location.href='/approve'} className="uditha-notify-button blink">Updates are Pending</button>

            </div>

        </div>

    )
}

export default Reviewers;