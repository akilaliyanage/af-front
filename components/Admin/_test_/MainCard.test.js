import React from 'react';
import {render, screen, cleanup} from '@testing-library/react';
import MainCard from '../MainCard';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import expectExport from 'expect';


afterEach(()=>cleanup());

it('Compare the snapshot of MainCard', () =>{
    const tree = renderer.create(
        <MainCard/>
    ).toJSON();
    expectExport(tree).toMatchSnapshot();
  })

test('Make sure Component render without any issue', () => {
    render(<MainCard />);
    screen.debug();
    expect(screen.getByText('Updates are Pending')).toBeInTheDocument();
  });

