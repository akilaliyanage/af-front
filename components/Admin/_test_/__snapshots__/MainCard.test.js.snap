// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Compare the snapshot of MainCard 1`] = `
<div
  className="uditha-main-card uditha-main-card-align"
>
  <div
    className="uditha-edit-content"
  >
    <div
      className="uditha-main-card-title"
    >
      <h1 />
    </div>
    <br />
    <div
      className="uditha-main-card-title"
    >
      <h3 />
    </div>
    <br />
    <h2>
      @
    </h2>
    <h1>
      SLIIT
    </h1>
    <br />
    <button
      className="uditha-notify-button blink"
      onClick={[Function]}
      style={
        Object {
          "display": "",
        }
      }
    >
      Updates are Pending
    </button>
  </div>
</div>
`;

exports[`create a snapshot of MainCard 1`] = `
<div
  className="uditha-main-card uditha-main-card-align"
>
  <div
    className="uditha-edit-content"
  >
    <div
      className="uditha-main-card-title"
    >
      <h1 />
    </div>
    <br />
    <div
      className="uditha-main-card-title"
    >
      <h3 />
    </div>
    <br />
    <h2>
      @
    </h2>
    <h1>
      SLIIT
    </h1>
    <br />
    <button
      className="uditha-notify-button blink"
      onClick={[Function]}
      style={
        Object {
          "display": "",
        }
      }
    >
      Updates are Pending
    </button>
  </div>
</div>
`;
